models/evc.py

```diff
@@ -1201,51 +1198,34 @@ def check_list_traces(list_circuits):
         if not list_circuits:
             return {}
         uni_list = []
-        circuit_data = {}
-        for circuit_id in list_circuits:
-            circuit = list_circuits[circuit_id]
+        for circuit in list_circuits:
             # if a inter-switch EVC does not have current_path, it does not
             # make sense to run sdntrace on it
             if not circuit.is_intra_switch() and not circuit.current_path:
+                list_circuits.remove(circuit)
                 continue
             uni_list.append(circuit.uni_a)
             uni_list.append(circuit.uni_z)
-            interface_a = uni_to_str(circuit.uni_a)
-            circuit_data[interface_a] = {
-                                            'circuit_id': circuit.id,
-                                            'trace_name': 'trace_a'
-                                        }
-            interface_z = uni_to_str(circuit.uni_z)
-            circuit_data[interface_z] = {
-                                            'circuit_id': circuit.id,
-                                            'trace_name': 'trace_z'
-                                        }
 
         traces = EVCDeploy.run_bulk_sdntraces(uni_list)
+        traces = traces["result"]
 
-        circuit_by_traces = {}
         circuits_checked = {}
 
-        for trace_switch in traces:
-            for trace in traces[trace_switch]:
```

Comment:
Nicely done, look how simpler it became once the right data structure is in place to reason about and maintain the code.

Reply:
Thank you, that is.